src/NuGet.Core/NuGet.Test.Utility/TestDirectory.cs

```diff
@@ -17,15 +17,23 @@ public TestDirectory(string path)
 
         public void Dispose()
         {
-            TestFileSystemUtility.AssertNotTempPath(Path);
+            TestFileSystemUtility.DeleteRandomTestFolder(Path);
+        }
 
-            try
-            {
-                Directory.Delete(Path, recursive: true);
-            }
-            catch
+        public static TestDirectory Create(string path = null)
+        {
+            var randomFolderName = Guid.NewGuid().ToString();
+
+            path = string.IsNullOrWhiteSpace(path) ? System.IO.Path.Combine(TestFileSystemUtility.NuGetTestFolder, randomFolderName) : path;
+
+            if (Directory.Exists(path))
             {
+                throw new InvalidOperationException("Guid collision");
```

Comment:
There was a time when these temp folders weren't behaving to the point where this paranoia almost made sense 😆